src/main/java/com/puppycrawl/tools/checkstyle/PackageObjectFactory.java

```diff
@@ -91,7 +97,7 @@ private Object doMakeObject(String name)
                 return createObject(className);
             }
             catch (final CheckstyleException ex) {
-                // keep looking
+                LOG.debug("Keep looking, ignoring exception", ex);
```

Comment:
By merging those two blocks you can increase code coverage and (based on cobertura output) cover this fully.